internal/tiltfile/telemetry/telemetry.go

```diff
@@ -0,0 +1,80 @@
+package telemetry
+
+import (
+	"fmt"
+
+	"go.starlark.net/starlark"
+
+	"github.com/windmilleng/tilt/internal/tiltfile/starkit"
+	"github.com/windmilleng/tilt/pkg/model"
+)
+
+type Settings struct {
```

Comment:
fwiw, I've been defining these sorts of settings objects in pkg/model, and handing them directly to EngineState, rather than make tiltfileLoader do custom unpacking logic per-model

I think this starts to point to a general Tilt extension model where you define a (1) Starkit extension, (2) a data model, and (3) a subscriber to that model that kinda parallels the CRD spec / status / operator mindset